### DISCUSSION TITLE
Only on Raspberry Pi 0??

### DISCUSSION THREAD

**Discussion by issinho2006** [lcintron/star-wars-ge-droid-pod#1] [Q&A] [answered]
You have this working on a Raspberry Pi 0, what about a RPi 3 or even 4??

**Comment by lcintron** [comment, marked as answer]
It should be no different.

**Reply by issinho2006** [reply]
Without the added screen that you mention, I'm not seeing it working.  I mean, script seems to run, but no reaction and I'm wondering if it's due to not being able to select options without the screen addon.

**Reply by lcintron** [reply]
The ui service drives the advertisment of the signals based on what's selected. To bypass that, you can just run the nodejs server which controls the BLE module and call the api endpoint at `/start/EIRAdvertisement/<data>`  where <data> is a _data value_ found in the [data.json](https://github.com/lcintron/star-wars-ge-droid-pod/blob/master/ui/data.json) file. 

In essence, all you need to run is `/usr/local/bin/node /home/pi/star-wars-ge-droid-pod/server/index.js &` and drive the signals being advertised over http with GET requests.

**Reply by issinho2006** [reply]
I'm sorry to keep bugging you with this.  I'm trying to start it up with "node ~/star-wars-ge-droid-pod/server/index.js" and getting this:

 /usr/local/bin/node /home/pi/star-wars-ge-droid-pod/server/index.js
events.js:165
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::3000
    at Server.setupListenHandle [as _listen2] (net.js:1346:14)
    at listenInCluster (net.js:1387:12)
    at Server.listen (net.js:1475:7)
    at Function.listen (/home/pi/star-wars-ge-droid-pod/server/node_modules/express/lib/application.js:618:24)
    at Object.<anonymous> (/home/pi/star-wars-ge-droid-pod/server/index.js:9:5)
    at Module._compile (internal/modules/cjs/loader.js:654:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:665:10)
    at Module.load (internal/modules/cjs/loader.js:566:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:506:12)
    at Function.Module._load (internal/modules/cjs/loader.js:498:3)
Emitted 'error' event at:
    at emitErrorNT (net.js:1366:8)
    at process._tickCallback (internal/process/next_tick.js:178:19)
    at Function.Module.runMain (internal/modules/cjs/loader.js:697:11)
    at startup (internal/bootstrap/node.js:201:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:516:3)

Also, please forgive me, I'm not entirely sure what you're getting at with "drive the signals being advertised over https with GET requests".

I am super excited to try to get this to work and your project seems to be the only option I've found that doesn't require to go an try and find one of those micro:bit boards (I'd REALLY like to try and get this to work with what I have).

**Reply by lcintron** [reply]
That error means that the webserver is already running on the pi. You can open up your browser and type localhost:3000/ or via curl to see the status of the BLE signal veing advertised. To broadcast a signal from the data json file, go to localhost:3000/start/EIRAdvertisement/<data> and replace <data> with the data field value if the signal you want to advertised. You can then confirm whether that's being advertised by the pi using a BLE monitoring app such as nRF on Android. I think the astromech droids react to these every 1 or 2 minutes. So not always immediately.

**Comment by issinho2006** [comment]
So, just to be clear:  <IP of
PI>:3000/start/EIRAdvertisement/02010609FF83010a040102a601060944524F4944,
to start Resistance Area, correct??

That just shows spinning, which is why I'm trying to verify.  Leaving off
the hex code gives me:

Cannot GET /start/EIRAdvertisement.

Also, doing a curl against localhost:3000, I see this:

{"status":"success","data":{"platform":"linux","state":"poweredOff","advertising":false,"address":"unknown","rssi":0,"mtu":20,"eirData":""}}

The "poweredOff" state is what's making me wonder if there's something off
on my install

On Thu, Jun 9, 2022 at 11:02 PM Luis Cintron ***@***.***>
wrote:

> That error means that the webserver is already running on the pi. You can
> open up your browser and type localhost:3000/ or via curl to see the status
> of the BLE signal veing advertised. To broadcast a signal from the data
> json file, go to localhost:3000/start/EIRAdvertisement/ and replace with
> the data field value if the signal you want to advertised. You can then
> confirm whether that's being advertised by the pi using a BLE monitoring
> app such as nRF on Android. I think the astromech droids react to these
> every 1 or 2 minutes. So not always immediately.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lcintron/star-wars-ge-droid-pod/discussions/1#discussioncomment-2919373>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AW5PQT53LMOSNLRZAGNICT3VOLD5FANCNFSM5YIREBHA>
> .
> You are receiving this because you authored the thread.Message ID:
> <lcintron/star-wars-ge-droid-pod/repo-discussions/1/comments/2919373@
> github.com>
>


-- 
Jeremy Hamilton
***@***.***

"Microsoft is not the answer. Microsoft is the question. NO is the
answer."
-- Erik Naggum

"Nothing is more difficult, and therefore more precious, than to be able to
decide."
--Napoleon Bonaparte

**Reply by issinho2006** [reply]
Fiddled around with this a bit more.  Decided to try and change the port that the server listens on.  Seemed to work for a bit, other than the droid not cooperating.  still not sure how to get this working.

**Comment by issinho2006** [comment]
and just to add:

 curl
localhost:3000/start/EIRAdvertisement/02010609FF83010a040702a601060944524F4944
curl: (52) Empty reply from server

On Fri, Jun 10, 2022 at 10:26 AM Jeremy Hamilton ***@***.***>
wrote:

> So, just to be clear:  <IP of
> PI>:3000/start/EIRAdvertisement/02010609FF83010a040102a601060944524F4944,
> to start Resistance Area, correct??
>
> That just shows spinning, which is why I'm trying to verify.  Leaving off
> the hex code gives me:
>
> Cannot GET /start/EIRAdvertisement.
>
> Also, doing a curl against localhost:3000, I see this:
>
>
> {"status":"success","data":{"platform":"linux","state":"poweredOff","advertising":false,"address":"unknown","rssi":0,"mtu":20,"eirData":""}}
>
> The "poweredOff" state is what's making me wonder if there's something off
> on my install
>
> On Thu, Jun 9, 2022 at 11:02 PM Luis Cintron ***@***.***>
> wrote:
>
>> That error means that the webserver is already running on the pi. You can
>> open up your browser and type localhost:3000/ or via curl to see the status
>> of the BLE signal veing advertised. To broadcast a signal from the data
>> json file, go to localhost:3000/start/EIRAdvertisement/ and replace with
>> the data field value if the signal you want to advertised. You can then
>> confirm whether that's being advertised by the pi using a BLE monitoring
>> app such as nRF on Android. I think the astromech droids react to these
>> every 1 or 2 minutes. So not always immediately.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/lcintron/star-wars-ge-droid-pod/discussions/1#discussioncomment-2919373>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AW5PQT53LMOSNLRZAGNICT3VOLD5FANCNFSM5YIREBHA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <lcintron/star-wars-ge-droid-pod/repo-discussions/1/comments/2919373@
>> github.com>
>>
>
>
> --
> Jeremy Hamilton
> ***@***.***
>
> "Microsoft is not the answer. Microsoft is the question. NO is the
> answer."
> -- Erik Naggum
>
> "Nothing is more difficult, and therefore more precious, than to be able
> to decide."
> --Napoleon Bonaparte
>


-- 
Jeremy Hamilton
***@***.***

"Microsoft is not the answer. Microsoft is the question. NO is the
answer."
-- Erik Naggum

"Nothing is more difficult, and therefore more precious, than to be able to
decide."
--Napoleon Bonaparte
